pkg/plugins/datasource_plugin.go

```diff
@@ -60,12 +60,17 @@ var handshakeConfig = plugin.HandshakeConfig{
 func (p *DataSourcePlugin) startBackendPlugin(ctx context.Context, log log.Logger) error {
 	p.log = log.New("plugin-id", p.Id)
 
-	err := p.spawnSubProcess()
-	if err == nil {
-		go p.restartKilledProcess(ctx)
+	if err := p.spawnSubProcess(); err != nil {
+		return err
 	}
 
-	return err
+	go func() {
+		if err := p.restartKilledProcess(ctx); err != nil {
+			p.log.Error("Attempting to restart killed process failed", "err", err)
```

Comment:
Probably make sense to check for `context.Cancelled` error here since `restartKilledProcess` returns `ctx.Err()` and not log if the context is cancelled. Does that make sense to you?